chronos/asyncmacro2.nim

```diff
@@ -247,13 +290,23 @@ proc asyncSingleProc(prc: NimNode): NimNode {.compileTime.} =
     #      against the spirit of the raises annotation - one should investigate
     #      here the possibility of transporting more specific error types here
     #      for example by casting exceptions coming out of `await`..
-    closureIterator.addPragma(nnkExprColonExpr.newTree(
-      newIdentNode("raises"),
-      nnkBracket.newTree(
-        newIdentNode("Defect"),
-        newIdentNode("CatchableError")
-      )
-    ))
+    when defined(chronosStrictException):
+      closureIterator.addPragma(nnkExprColonExpr.newTree(
+        newIdentNode("raises"),
+        nnkBracket.newTree(
+          newIdentNode("Defect"),
+          newIdentNode("CatchableError")
+        )
+      ))
+    else:
+      closureIterator.addPragma(nnkExprColonExpr.newTree(
```

Comment:
Doesn't `Exception` make all `Defect` and `CatchableError` reduntant?

Reply:
sort of, but it also don't hurt - ie it's easier to see what the hack is this way.